[Feature] Add `update_repo_settings` function to HfApi #2447 (#2502)
huggingface/huggingface_hub: docs/source/en/guides/repository.md

>>> update_repo_visibility(repo_id=repo_id, private=True)
```

> ### Update repository settings (Gated Access)

Reviewer: ```suggestion
### Setup gated access
```

(this section is already under `## Change repository settings`)

> The `update_repo_settings` function allows you to control the gated access feature of a repository. Gated access restricts access to the repository's files, requiring users to request access before they can view or download the content.
> You can update the settings of a repository, using the `update_repo_settings` function as shown in the following:

Reviewer: ```suggestion
To give more control over how repos are used, the Hub allows repo authors to enable **access requests** for their repos. Users must agree to share their contact information (username and email address) with the repo authors to access the  files when enabled. A repo with access requests enabled is called a **gated repo**.

You can set a repo as gated using [`update_repo_settings`]:
```

I took over the existing wording from https://huggingface.co/docs/hub/models-gated introduction to be more consistent in our wording.

> **Note:** This method is currently designed to work primarily with **dataset** repositories.
> 

Reviewer: ```suggestion
```

Except very rare exceptions, methods of `HfApi` must be primarily designed to work on model repos.

```py
>>> from huggingface_hub import HfApi

>>> api = HfApi()

> >>> api.update_repo_settings(repo_id=repo_id, gated="auto")  # Set automatic gating for a dataset

Reviewer: ```suggestion
>>> api.update_repo_settings(repo_id=repo_id, gated="auto")  # Set automatic gating for a model
```
